packages/server/src/db/index.js

```diff
@@ -647,26 +661,76 @@ async function getAgenciesByIds(agencyIds) {
     return result;
 }
 
-async function getAgenciesSubscribedToDigest() {
-    const query = knex
-        .select(
-            'agencies.id',
-            'agencies.name',
-            'agencies.abbreviation',
-            'agencies.code',
+async function getAgenciesSubscribedToDigest(asOf) {
+    const open_date = asOf || moment().subtract(1, 'day').format('YYYY-MM-DD');
+
+    const query = knex.raw(
+        `
+        WITH enabled_codes AS (
+            SELECT
+                a.id AS agency_id,
+                ec.code,
+                CASE WHEN aec.enabled IS NULL THEN
+                    TRUE
+                ELSE
+                    aec.enabled
+                END as enabled
+            FROM
+                eligibility_codes ec
+            CROSS JOIN agencies a
+            LEFT JOIN agency_eligibility_codes aec ON ec.code = aec.code
+                AND a.id = aec.agency_id
+        ),
+        agency_data AS (
+            SELECT
+                a.id,
+                a.name,
+                string_agg(DISTINCT aec.code,
+                    '|') AS codes,
+                string_agg(DISTINCT k.search_term,
+                    '|') AS term,
+                string_agg(DISTINCT u.email,
+                    ' | ') AS emails
+            FROM
+                agencies a
+            JOIN enabled_codes aec ON aec.agency_id = a.id
+                AND aec.enabled = TRUE
+            JOIN keywords k ON k.agency_id = a.id
+            JOIN users u ON u.agency_id = a.id
+        GROUP BY
+            a.id
         )
-        .from('agencies')
-        .join('keywords', 'keywords.agency_id', '=', 'agencies.id')
-        .orderBy('agencies.id')
-        .groupBy(
-            'agencies.id',
-            'agencies.name',
-            'agencies.abbreviation',
-            'agencies.code',
-        );
+        SELECT
+            ad.id,
+            ad.name,
+            ad.emails AS user_emails,
+            string_agg(DISTINCT g.grant_id, '|') AS matched_grant_ids
+        FROM
+            grants g
+            JOIN agency_data ad ON g.eligibility_codes ~ ad.codes
```

Comment:
Just a note in case you end up looking into `array_agg` – this `ON` clause would need to be replaced with something like this:
```sql
JOIN agency_data ad ON string_to_array(g.eligibility_codes, ' ') && ad.codes
```
which would check whether the space-split array of `g.eligibility_codes` values has elements in common with `ad.codes`.

Reply:
I decided to do `array_to_string(ad.codes, '|')` here since its consistent with our existing keyword and eligibility code searching logic. wdyt?